crypto/fipsmodule/rand/urandom_test.cc

```diff
@@ -207,6 +217,7 @@ static void GetTrace(std::vector<Event> *out_trace, unsigned flags,
     struct user_regs_struct regs;
     ASSERT_EQ(0, ptrace(PTRACE_GETREGS, child_pid, nullptr, &regs));
     const auto syscall_number = regs.orig_rax;
+    static auto previous_syscall = regs.orig_rax;
```

Comment:
Since this is inside the for loop it makes `previous_syscall` basically always equal to `syscall_number`

Reply:
Not sure I can see the issue. Can you explain your concern more?

`previous_syscall` tracks the previous system call such that we don't account for more than one `__NR[_clock_]nanosleep`.

Reply:
You set `previous_syscall` to the current syscall every loop

Reply:
Solved offline, I missed that static there